### DISCUSSION TITLE
List of APIs

### DISCUSSION THREAD

**Discussion by jonas-kupferschmid** [bitcraze/discussions#699] [Q&A]
Is there a list of all the APIs that are supported and how they can be used in the APP layer? Or just some important ones for me to look out for, as https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/params/ isn't too well documented for me.

Some really useful for me would be clearer documentation of 
1. How to do Setpoints
2. A way to make the controller react faster (more aggressive) 
3. setting individual thrust values for open loop behaviours

And maybe a quick guide on how to create custom APIs

(my questions are concerning the crazyflie firmware in C and cfclient)
Thank you!

**Comment by ataffanel** [comment]
Hi,

There is no good list of API usable for app yet unfortunatly. The best we have so far is the [app_api](https://github.com/bitcraze/crazyflie-firmware/blob/master/app_api/src/app_main.c), this is an APP that calls all the API we consider public and stable for app to use. This app is not doing anything useful, but it is build automatically in CI so if we inadvertently break those function we will know about it and fix the problem right away.

So all the functions used in the `app_api` app are good to use in an app. Off course, C being an old language, there is no fine-grained protection so everything can be called from an app, there is just no guarantee or checks that the other function will be stable over time.

As for adding new API, it depends of the sense of your questions. In your own projects you can add as many sub-repos or lib as you want/need. In the Crazyflie firmware repos, a public APP API is one that is in the `app_api` app, so adding a new function there is done by pull-request to the Crazyflie firmware repos.
